
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A Nextflow-based DNA Methylation Data Analysis Pipeline for Twist NGS data">
      
      
        <meta name="author" content="Jyotirmoy Das">
      
      
      
        <link rel="prev" href="../overview/">
      
      
        <link rel="next" href="../faqs/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.18">
    
    
      
        <title>Usage - TwistMethylFlow</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.7e37652d.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Inter:300,300i,400,400i,700,700i%7CFira+Code:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Inter";--md-code-font:"Fira Code"}</style>
      
    
    
      <link rel="stylesheet" href="../styles/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="blue-grey" data-md-color-accent="blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#computational-requirements" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="TwistMethylFlow" class="md-header__button md-logo" aria-label="TwistMethylFlow" data-md-component="logo">
      

<a href="https://github.com/JD2112" target="_blank" style="display:flex; align-items:center; text-decoration:none;">
  <img src="../images/JD_logo.png" alt="Logo" style="height:24px; margin-right:8px;">
</a>
    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            TwistMethylFlow
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Usage
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/JD2112/TwistMethylFlow" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../overview/" class="md-tabs__link">
        
  
  
    
  
  Description

      </a>
    </li>
  

      
        
  
  
  
    
  
  
    <li class="md-tabs__item md-tabs__item--active">
      <a href="./" class="md-tabs__link">
        
  
  
    
  
  Usage

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../faqs/" class="md-tabs__link">
        
  
  
    
  
  FAQs

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../references/" class="md-tabs__link">
        
  
  
    
  
  References

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="TwistMethylFlow" class="md-nav__button md-logo" aria-label="TwistMethylFlow" data-md-component="logo">
      

<a href="https://github.com/JD2112" target="_blank" style="display:flex; align-items:center; text-decoration:none;">
  <img src="../images/JD_logo.png" alt="Logo" style="height:24px; margin-right:8px;">
</a>
    </a>
    TwistMethylFlow
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/JD2112/TwistMethylFlow" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
  
  <li class="md-nav__item">
    
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
    
    
    
  </li>

  
  <li class="md-nav__item">
    
  
  
  
  
    <li class="md-nav__item">
      <a href="../overview/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Description
    
  </span>
  

      </a>
    </li>
  

    
    
    
    
  </li>

  
  <li class="md-nav__item">
    
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Usage
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Usage
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#computational-requirements" class="md-nav__link">
    <span class="md-ellipsis">
      Computational Requirements
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Computational Requirements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#softwares-requirements" class="md-nav__link">
    <span class="md-ellipsis">
      Software/s Requirements
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hardware-requirements" class="md-nav__link">
    <span class="md-ellipsis">
      Hardware Requirements
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-nextflow" class="md-nav__link">
    <span class="md-ellipsis">
      Run NextFlow
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parameters-configuration" class="md-nav__link">
    <span class="md-ellipsis">
      Parameters configuration
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#input-file" class="md-nav__link">
    <span class="md-ellipsis">
      Input File
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#results" class="md-nav__link">
    <span class="md-ellipsis">
      Results
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
    
    
    
  </li>

  
  <li class="md-nav__item">
    
  
  
  
  
    <li class="md-nav__item">
      <a href="../faqs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    FAQs
    
  </span>
  

      </a>
    </li>
  

    
    
    
    
  </li>

  
  <li class="md-nav__item">
    
  
  
  
  
    <li class="md-nav__item">
      <a href="../references/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    References
    
  </span>
  

      </a>
    </li>
  

    
    
    
    
  </li>


  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



  <h1>Usage</h1>

<p>The pipeline needs to use NextFlow, Singularity/Docker/Conda (for container). Details are given in corresponding pages.</p>
<ul>
<li>It is tested locally on Ubuntu server 24.04.2 LTS.</li>
<li>Tested on HPC (Dardel from PDC, KTH)</li>
</ul>
<h2 id="computational-requirements">Computational Requirements<a class="headerlink" href="#computational-requirements" title="Permanent link">&para;</a></h2>
<h3 id="softwares-requirements">Software/s Requirements<a class="headerlink" href="#softwares-requirements" title="Permanent link">&para;</a></h3>
<ul>
<li><a href="https://www.nextflow.io/docs/latest/install.html#install-nextflow">Nextflow (&gt;=21.10.3)</a></li>
<li><a href="https://docs.docker.com/engine/install/">Docker</a> or <a href="https://singularity-tutorial.github.io/01-installation/">Singularity</a> (for containerized execution)</li>
<li>Java (&gt;=8)</li>
</ul>
<h3 id="hardware-requirements">Hardware Requirements<a class="headerlink" href="#hardware-requirements" title="Permanent link">&para;</a></h3>
<ul>
<li>RAM: 6 GB - 200 GB</li>
<li>CPU: min. 12 cores</li>
<li>Storage: ~2TB for 24 paired-end samples</li>
</ul>
<h2 id="run-nextflow">Run NextFlow<a class="headerlink" href="#run-nextflow" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>nextflow run JD2112/TwistMethylFlow \
    -profile singularity \
    --sample_sheet Sample_sheet_twist.csv \
    --genome_fasta /data/reference_genome/hg38/hg38.fa \ 
    --run_both_methods \
    --gtf_file /data/Homo_sapiens.GRCh38.104.gtf \
    --refseq_file /data/hg38_RefSeq.bed.gz \
    --outdir Results/TwistMethylFlow_both 
</code></pre></div>
<details class="info">
<summary>additional options</summary>
<p><strong>Run with pre-build reference genome index</strong>
<code>--bismark_index /data/reference_genome/hg38/</code></p>
<p><strong>Run only EdgeR</strong>
<code>--diff_meth_method edger</code></p>
<p><strong>Run only MethylKit</strong>
<code>--diff_meth_method methylkit</code></p>
</details>
<details class="warning">
<summary>need annotation files</summary>
<p>Remember to add annotation files for different differential methylation analysis</p>
<p><code>--gtf_file /data/Homo_sapiens.GRCh38.104.gtf</code> for <strong>MethylKit</strong> </p>
<p><code>--refseq_file /data/hg38_RefSeq.bed.gz</code> for <strong>EdgeR</strong></p>
</details>
<details class="note">
<summary>Help</summary>
<div class="highlight"><pre><span></span><code>nextflow run main.nf --help --outdir .
</code></pre></div>
</details>
<h2 id="parameters-configuration">Parameters configuration<a class="headerlink" href="#parameters-configuration" title="Permanent link">&para;</a></h2>
<p>User can change the <code>conf/params.config</code> or use the <code>--</code> flags directly on nextflow command line.</p>
<table>
<thead>
<tr>
<th>options</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--sample_sheet</code></td>
<td>Path to the sample sheet CSV file (required)</td>
</tr>
<tr>
<td><code>--bismark_index</code></td>
<td>Path to the Bismark index directory (required unless <code>--genome</code> or <code>--aligned_bams</code> is provided)</td>
</tr>
<tr>
<td><code>--genome</code></td>
<td>Path to the reference genome FASTA file (required if <code>--bismark_index</code> not provided)</td>
</tr>
<tr>
<td><code>--aligned_bams</code></td>
<td>Path to aligned BAM files (use this to start from aligned BAM files instead of FASTQ files)</td>
</tr>
<tr>
<td><code>--refseq_file</code></td>
<td>Path to RefSeq file for annotation (reuired to run <code>both</code> or <code>methylkit</code>)</td>
</tr>
<tr>
<td><code>--gtf_file</code></td>
<td>Path to GTF file for annotation (reuired to run <code>both</code> or <code>edger</code>)</td>
</tr>
<tr>
<td><code>--outdir</code></td>
<td>Output directory (default: ./results)</td>
</tr>
<tr>
<td><code>--diff_meth_method</code></td>
<td>Differential methylation method to use: 'edger' or 'methylkit' (default: edger)</td>
</tr>
<tr>
<td><code>--run_both_methods</code></td>
<td>Run both edgeR and methylkit for differential methylation analysis (default: false)</td>
</tr>
<tr>
<td><code>--skip_diff_meth</code></td>
<td>Skip differential methylation analysis (default: false)</td>
</tr>
<tr>
<td><code>--coverage_threshold</code></td>
<td>Minimum read coverage to consider a CpG site (default: 10)</td>
</tr>
<tr>
<td><code>--logfc_cutoff</code></td>
<td>Differential methylation cut-off for Volcano or MA plot (default: 1.5)</td>
</tr>
<tr>
<td><code>--pvalue_cutoff</code></td>
<td>Differential methylation P-value cut-off for Volcano or MA plot (default: 0.05)</td>
</tr>
<tr>
<td><code>--hyper_color</code></td>
<td>Hypermethylation color for Volcano or MA plot (default: red)</td>
</tr>
<tr>
<td><code>--hypo_cutoff</code></td>
<td>Hypomethylation color for Volcano or MA plot (default: blue)</td>
</tr>
<tr>
<td><code>--nonsig_color</code></td>
<td>Non-significant color for Volcano or MA plot (default: black)</td>
</tr>
<tr>
<td><code>--compare_str</code></td>
<td>Comparison string for differential analysis (e.g. "Group1-Group2")</td>
</tr>
<tr>
<td><code>--top_n_genes</code></td>
<td>Number of top differentially methylated genes to report for GOplot (default: 100)</td>
</tr>
<tr>
<td><code>--help</code></td>
<td>Show this help message and exit</td>
</tr>
</tbody>
</table>
<details class="info">
<summary>MethylKit specific parameters</summary>
<p><code>--assembly</code> - user needs to provide the genome assembly version. Default: <code>hg38</code></p>
<p><code>--mc_cores</code> - if required to run on multiple cores. Default: <code>1</code></p>
<p><code>--diff</code> - Differential methylation cutoff value. Default: <code>0.5</code></p>
<p><code>--qvalue</code> - qvalue respect to differential methylation value to identify significant CpGs. Default: <code>0.05</code></p>
</details>
<details class="note">
<summary>Other parameters configuration</summary>
<ol>
<li><code>conf/resource.config</code> - for resource settings.</li>
<li><code>conf/base.config</code> - for base settings.</li>
<li><code>nextflow.config</code> - for nextflow run with default setting.</li>
<li><code>dag.config</code> - for DAG configuration settings.</li>
</ol>
</details>
<h2 id="input-file">Input File<a class="headerlink" href="#input-file" title="Permanent link">&para;</a></h2>
<p>Sample sheet (CSV format) with sample information</p>
<p><code>Sample_sheet.csv</code>:</p>
<table>
<thead>
<tr>
<th>sample_id</th>
<th>group</th>
<th>read1</th>
<th>read2</th>
</tr>
</thead>
<tbody>
<tr>
<td>SN09</td>
<td>Healthy</td>
<td>FASTQ/SN09/SL1_S9_R1_001.fastq.gz</td>
<td>FASTQ/SN09/SL1_S9_R2_001.fastq.gz</td>
</tr>
<tr>
<td>SN10</td>
<td>Disease</td>
<td>FASTQ/SN10/SL2_S10_R1_001.fastq.gz</td>
<td>FASTQ/SN10/SL2_S10_R2_001.fastq.gz</td>
</tr>
<tr>
<td>SN11</td>
<td>Healthy</td>
<td>FASTQ/SN11/SL3_S11_R1_001.fastq.gz</td>
<td>FASTQ/SN11/SL3_S11_R2_001.fastq.gz</td>
</tr>
<tr>
<td>SN12</td>
<td>Disease</td>
<td>FASTQ/SN12/SL4_S12_R1_001.fastq.gz</td>
<td>FASTQ/SN12/SL4_S12_R2_001.fastq.gz</td>
</tr>
<tr>
<td>SN13</td>
<td>Healthy</td>
<td>FASTQ/SN13/SL5_S13_R1_001.fastq.gz</td>
<td>FASTQ/SN13/SL5_S13_R2_001.fastq.gz</td>
</tr>
<tr>
<td>SN14</td>
<td>Disease</td>
<td>FASTQ/SN14/SL6_S14_R1_001.fastq.gz</td>
<td>FASTQ/SN14/SL6_S14_R2_001.fastq.gz</td>
</tr>
</tbody>
</table>
<h2 id="results">Results<a class="headerlink" href="#results" title="Permanent link">&para;</a></h2>
<p>Here is how the result folder looks like -</p>
<div class="highlight"><pre><span></span><code>.
├── annotate_results
├── bismark_align
├── bismark_deduplicate
├── bismark_genome_preparation
├── bismark_methylation_extractor
├── bismark_report
├── edger_analysis
├── fastqc
├── go_analysis
├── multiqc
├── pipeline_info
├── post_processing
├── qualimap
├── samtools_index
├── samtools_sort
└── trim_galore
</code></pre></div>
<details class="info">
<summary>bismark alignment with bowtie2</summary>
<table>
<thead>
<tr>
<th>Option</th>
<th>Functionality</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>-q</code></td>
<td>Quiet mode: suppresses detailed output.</td>
</tr>
<tr>
<td><code>--score-min L,0,-0.2</code></td>
<td>Sets a linear minimum score for valid alignments (moderate stringency).</td>
</tr>
<tr>
<td><code>--ignore-quals</code></td>
<td>Ignores base quality scores during alignment.</td>
</tr>
<tr>
<td><code>--no-mixed</code></td>
<td>Ensures both ends of paired reads align properly; no single-end alignments.</td>
</tr>
<tr>
<td><code>--no-discordant</code></td>
<td>Prevents discordant alignments; enforces proper orientation and distance.</td>
</tr>
<tr>
<td><code>--dovetail</code></td>
<td>Allows overlapping or extended alignments in paired-end reads.</td>
</tr>
<tr>
<td><code>--maxins 500</code></td>
<td>Sets the maximum allowed distance between paired-end reads to 500 bases.</td>
</tr>
</tbody>
</table>
</details>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
  <div style="
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
    font-size: 1.5em;
    padding: 1em 2em;
    border-top: 1px solid #000000;
    color: #ffffff;
    text-align: center;
    background-color: #000000;
  ">
    <div style="flex: 1; text-align: left;">
        © Jyotirmoy Das 2025 
    </div>
    <div style="flex: 1; text-align: center;">
      <code style="font-size: 0.8em;">
        3D1F C87F 8D52 BFD4 5A1D <br> 3C86 63F2 2E14 6A0A 6B98
      </code>
    </div>
    <div style="flex: 1; text-align: right;">
      Made with <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" style="color: #ffffff;">MkDocs</a>
    </div>
  </div>

    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["navigation.instant", "navigation.tracking", "navigation.expand", "navigation.sections", "navigation.top", "content.code.annotate", "content.tabs.link", "navigation.tabs", "search.suggest", "search.highlight", "toc.integrate", "content.action.edit", "content.code.copy", "versioning", "icons"], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../assets/javascripts/bundle.92b07e13.min.js"></script>
      
        <script src="https://fonts.googleapis.com/css2?family=Fira+Code&amp;family=Inter:wght@400;600&amp;display=swap"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js?config=TeX-MML-AM_CHTML"></script>
      
    
  </body>
</html>