
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A Nextflow-based DNA Methylation Data Analysis Pipeline for Twist NGS data">
      
      
        <meta name="author" content="Jyotirmoy Das">
      
      
      
        <link rel="prev" href="../usage/">
      
      
        <link rel="next" href="../references/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.18">
    
    
      
        <title>FAQs - TwistMethylFlow</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.7e37652d.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Inter:300,300i,400,400i,700,700i%7CFira+Code:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Inter";--md-code-font:"Fira Code"}</style>
      
    
    
      <link rel="stylesheet" href="../styles/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="blue-grey" data-md-color-accent="blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#faqs" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="TwistMethylFlow" class="md-header__button md-logo" aria-label="TwistMethylFlow" data-md-component="logo">
      

<a href="https://github.com/JD2112" target="_blank" style="display:flex; align-items:center; text-decoration:none;">
  <img src="../images/JD_logo.png" alt="Logo" style="height:24px; margin-right:8px;">
</a>
    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            TwistMethylFlow
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              FAQs
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/JD2112/TwistMethylFlow" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../overview/" class="md-tabs__link">
        
  
  
    
  
  Description

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../usage/" class="md-tabs__link">
        
  
  
    
  
  Usage

      </a>
    </li>
  

      
        
  
  
  
    
  
  
    <li class="md-tabs__item md-tabs__item--active">
      <a href="./" class="md-tabs__link">
        
  
  
    
  
  FAQs

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../references/" class="md-tabs__link">
        
  
  
    
  
  References

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="TwistMethylFlow" class="md-nav__button md-logo" aria-label="TwistMethylFlow" data-md-component="logo">
      

<a href="https://github.com/JD2112" target="_blank" style="display:flex; align-items:center; text-decoration:none;">
  <img src="../images/JD_logo.png" alt="Logo" style="height:24px; margin-right:8px;">
</a>
    </a>
    TwistMethylFlow
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/JD2112/TwistMethylFlow" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
  
  <li class="md-nav__item">
    
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
    
    
    
  </li>

  
  <li class="md-nav__item">
    
  
  
  
  
    <li class="md-nav__item">
      <a href="../overview/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Description
    
  </span>
  

      </a>
    </li>
  

    
    
    
    
  </li>

  
  <li class="md-nav__item">
    
  
  
  
  
    <li class="md-nav__item">
      <a href="../usage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usage
    
  </span>
  

      </a>
    </li>
  

    
    
    
    
  </li>

  
  <li class="md-nav__item">
    
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    FAQs
    
  </span>
  

      </a>
      
    </li>
  

    
    
    
    
  </li>

  
  <li class="md-nav__item">
    
  
  
  
  
    <li class="md-nav__item">
      <a href="../references/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    References
    
  </span>
  

      </a>
    </li>
  

    
    
    
    
  </li>


  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="faqs">FAQs<a class="headerlink" href="#faqs" title="Permanent link">&para;</a></h1>
<h1 id="twistmethylflow-faqs">TwistMethylFlow FAQs<a class="headerlink" href="#twistmethylflow-faqs" title="Permanent link">&para;</a></h1>
<h2 id="pipeline-steps">Pipeline Steps<a class="headerlink" href="#pipeline-steps" title="Permanent link">&para;</a></h2>
<h3 id="generate-reference-genome">Generate Reference Genome<a class="headerlink" href="#generate-reference-genome" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>What is the purpose of Generate Reference Genome?</summary>
<p>This step creates reference genome index files for <strong>Bismark</strong>, which are required for bisulfite read alignment.</p>
</details>
<h3 id="raw-data-qc">Raw Data QC<a class="headerlink" href="#raw-data-qc" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How is raw sequencing data quality assessed?</summary>
<p>Raw Data QC uses <strong>FastQC</strong> to evaluate per-base quality, GC content, adapter contamination, and other metrics.</p>
</details>
<h3 id="adapter-trimming">Adapter Trimming<a class="headerlink" href="#adapter-trimming" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How should adapters and low-quality bases be handled?</summary>
<p><strong>Trim Galore</strong> removes adapters and low-quality bases from reads, ensuring high-quality data for alignment.</p>
</details>
<h3 id="align-reads">Align Reads<a class="headerlink" href="#align-reads" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How are reads aligned to the reference genome?</summary>
<p><strong>Bismark (Bowtie2)</strong> aligns bisulfite-treated reads to the reference genome.<br />
Accurate alignment is crucial for downstream methylation analysis.</p>
</details>
<h3 id="deduplicate-removal">Deduplicate Removal<a class="headerlink" href="#deduplicate-removal" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>What is deduplicate removal?</summary>
<p><strong>Bismark</strong> removes PCR duplicates from aligned reads to prevent bias in methylation calling.</p>
</details>
<h3 id="sort-and-indexing">Sort and Indexing<a class="headerlink" href="#sort-and-indexing" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How are BAM files prepared after alignment?</summary>
<p><strong>Samtools</strong> sorts and indexes deduplicated BAM files for efficient access in downstream analyses.</p>
</details>
<h3 id="extract-methylation-calls">Extract Methylation Calls<a class="headerlink" href="#extract-methylation-calls" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How are methylation calls extracted?</summary>
<p><strong>Bismark</strong> extracts cytosine methylation calls from aligned reads, generating the input for differential methylation analysis.</p>
</details>
<h3 id="summary-report">Summary Report<a class="headerlink" href="#summary-report" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How is the summary report generated?</summary>
<p><strong>Bismark</strong> summarizes alignment statistics and methylation extraction results, including conversion efficiency.</p>
</details>
<h3 id="alignment-qc">Alignment QC<a class="headerlink" href="#alignment-qc" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How is alignment quality assessed?</summary>
<p><strong>Qualimap</strong> generates metrics such as coverage, mapping quality, and duplication rate for aligned reads.</p>
</details>
<h3 id="qc-reporting">QC Reporting<a class="headerlink" href="#qc-reporting" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How are QC reports aggregated?</summary>
<p><strong>MultiQC</strong> compiles QC reports from FastQC, Bismark, and Qualimap into a single comprehensive report.</p>
</details>
<h3 id="differential-methylation">Differential Methylation<a class="headerlink" href="#differential-methylation" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How is differential methylation analysis performed?</summary>
<p><strong>EdgeR</strong> and <strong>MethylKit</strong> identify differentially methylated positions or regions, considering replicates and experimental design.</p>
</details>
<h3 id="post-processing">Post Processing<a class="headerlink" href="#post-processing" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How are results visualized?</summary>
<p><strong>ggplot2</strong> creates summary plots like heatmaps, volcano plots, and coverage plots for differential methylation results.</p>
</details>
<h3 id="go-analysis">GO Analysis<a class="headerlink" href="#go-analysis" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How is Gene Ontology analysis performed?</summary>
<p>Functional enrichment analysis identifies pathways associated with differentially methylated genes, providing biological interpretation.</p>
</details>
<hr />
<h2 id="nextflow-workflow-concepts">Nextflow Workflow Concepts<a class="headerlink" href="#nextflow-workflow-concepts" title="Permanent link">&para;</a></h2>
<h3 id="general-nextflow">General Nextflow<a class="headerlink" href="#general-nextflow" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>What is Nextflow?</summary>
<p>Nextflow is a workflow management system for reproducible and scalable scientific pipelines.<br />
It allows users to define tasks and data flow using <strong>processes</strong> and <strong>channels</strong>.</p>
</details>
<h3 id="processes">Processes<a class="headerlink" href="#processes" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>What are processes in Nextflow?</summary>
<p>Processes encapsulate a task with defined inputs, outputs, and commands.<br />
They run independently and can scale across computing resources.</p>
</details>
<h3 id="channels">Channels<a class="headerlink" href="#channels" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>What are channels in Nextflow?</summary>
<p>Channels are asynchronous data streams that connect processes, passing input and output data between them.</p>
</details>
<h3 id="outputs">Outputs<a class="headerlink" href="#outputs" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How do I define outputs in Nextflow?</summary>
<p>Outputs are declared using the <code>output</code> block in processes or workflows.<br />
This defines what data is saved for downstream steps or exported from the workflow.</p>
</details>
<h3 id="parallel-execution">Parallel Execution<a class="headerlink" href="#parallel-execution" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How does Nextflow handle parallel execution?</summary>
<p>Nextflow automatically runs independent processes in parallel based on available compute resources, enabling scalable execution.</p>
</details>
<h3 id="reproducibility">Reproducibility<a class="headerlink" href="#reproducibility" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>How does Nextflow ensure reproducibility?</summary>
<p>Nextflow supports containerization (Docker/Singularity) and environment specification, ensuring the same pipeline produces identical results across systems.</p>
</details>
<h3 id="workflow-profiles">Workflow Profiles<a class="headerlink" href="#workflow-profiles" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>What are Nextflow profiles?</summary>
<p>Profiles allow users to define environment-specific configurations, such as compute clusters, Docker images, and resource limits, for flexible pipeline execution.</p>
</details>
<h3 id="learning-resources">Learning Resources<a class="headerlink" href="#learning-resources" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>Where can I learn more about Nextflow?</summary>
<ul>
<li><a href="https://www.nextflow.io/docs/latest/index.html">Nextflow Documentation</a>  </li>
<li><a href="https://nf-co.re/docs/usage/tutorials/nextflow">Nextflow Tutorials</a>  </li>
<li><a href="https://training.nextflow.io/2.0/hello_nextflow/">Hello Nextflow Training</a></li>
</ul>
</details>
<hr />
<h2 id="performance">Performance<a class="headerlink" href="#performance" title="Permanent link">&para;</a></h2>
<h3 id="runtime-memory-and-storage">Runtime, Memory, and Storage<a class="headerlink" href="#runtime-memory-and-storage" title="Permanent link">&para;</a></h3>
<details class="question">
<summary>What are the typical runtime, memory, and storage requirements?</summary>
<ul>
<li><strong>Runtime</strong>: Varies based on dataset size and computational resources. For 24 paired-end samples, it can take several hours to days.</li>
<li><strong>Memory</strong>: Ranges from 6 GB to 200 GB depending on the step. Alignment and differential methylation analysis are more memory-intensive.</li>
<li><strong>Storage</strong>: Approximately 2 TB for 24 paired-end samples, including intermediate files and results.</li>
</ul>
<table>
<thead>
<tr>
<th>Process</th>
<th>Average Process Time</th>
<th>Average Wall Clock Time</th>
<th>Max Peak Memory</th>
<th>Total I/O (Read + Written)</th>
</tr>
</thead>
<tbody>
<tr>
<td>FastQC</td>
<td>7m 3s</td>
<td>7m 2s</td>
<td>628.4 MB</td>
<td>5.3 GB</td>
</tr>
<tr>
<td>trim galore</td>
<td>22m 1s</td>
<td>22m 0s</td>
<td>348.2 MB</td>
<td>165.2 GB</td>
</tr>
<tr>
<td>Bismark Genome Preparation</td>
<td>2h 24m 38s</td>
<td>2h 24m 37s</td>
<td>12.8 GB</td>
<td>47.8 GB</td>
</tr>
<tr>
<td>Bismark Alignment</td>
<td>14h 27m 57s</td>
<td>14h 27m 56s</td>
<td>10.2 GB</td>
<td>229.8 GB</td>
</tr>
<tr>
<td>Bismark Deduplication</td>
<td>29m 3s</td>
<td>29m 1s</td>
<td>9.8 GB</td>
<td>95.8 GB</td>
</tr>
<tr>
<td>Samtools sort</td>
<td>8m 32s</td>
<td>8m 31s</td>
<td>3.3 GB</td>
<td>14.8 GB</td>
</tr>
<tr>
<td>Samtools index</td>
<td>1m 2s</td>
<td>1m 1s</td>
<td>21.1 MB</td>
<td>4.8 GB</td>
</tr>
<tr>
<td>Bismark Methylation extractor</td>
<td>2h 56m 23s</td>
<td>2h 56m 22s</td>
<td>2 GB</td>
<td>258.4 GB</td>
</tr>
<tr>
<td>Qualimap</td>
<td>6m 29s</td>
<td>6m 29s</td>
<td>12.1 GB</td>
<td>4.7 GB</td>
</tr>
<tr>
<td>Bismark report</td>
<td>2.9s</td>
<td>2.5s</td>
<td>161.1 MB</td>
<td>7.1 MB</td>
</tr>
<tr>
<td>Multiqc</td>
<td>14.1s</td>
<td>13.3s</td>
<td>80.5 MB</td>
<td>3.0 GB</td>
</tr>
<tr>
<td>EdgeR analysis</td>
<td>46m 37s</td>
<td>46m 36s</td>
<td>44.9 GB</td>
<td>8.9 GB</td>
</tr>
<tr>
<td>Annotate results</td>
<td>5m 23s</td>
<td>5m 23s</td>
<td>6.8 GB</td>
<td>1.8 GB</td>
</tr>
<tr>
<td>GO analysis EdgeR</td>
<td>4m 15s</td>
<td>4m 15s</td>
<td>6.4 GB</td>
<td>1.8 GB</td>
</tr>
<tr>
<td>Post processing EdgeR</td>
<td>18m 40s</td>
<td>18m 40s</td>
<td>9.3 GB</td>
<td>7.3 GB</td>
</tr>
<tr>
<td>MethylKit analysis</td>
<td>2h 59m 19s</td>
<td>2h 59m 19s</td>
<td>37.6 GB</td>
<td>9.0 GB</td>
</tr>
<tr>
<td>GO analysis methylKit</td>
<td>1m 44s</td>
<td>1m 44s</td>
<td>3.9 GB</td>
<td>1.3 GB</td>
</tr>
<tr>
<td>Post processing methylKit</td>
<td>8m 4s</td>
<td>8m 3s</td>
<td>4.4 GB</td>
<td>3.5 GB</td>
</tr>
</tbody>
</table>
</details>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
  <div style="
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
    font-size: 1.5em;
    padding: 1em 2em;
    border-top: 1px solid #000000;
    color: #ffffff;
    text-align: center;
    background-color: #000000;
  ">
    <div style="flex: 1; text-align: left;">
        © Jyotirmoy Das 2025 
    </div>
    <div style="flex: 1; text-align: center;">
      <code style="font-size: 0.8em;">
        3D1F C87F 8D52 BFD4 5A1D <br> 3C86 63F2 2E14 6A0A 6B98
      </code>
    </div>
    <div style="flex: 1; text-align: right;">
      Made with <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" style="color: #ffffff;">MkDocs</a>
    </div>
  </div>

    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["navigation.instant", "navigation.tracking", "navigation.expand", "navigation.sections", "navigation.top", "content.code.annotate", "content.tabs.link", "navigation.tabs", "search.suggest", "search.highlight", "toc.integrate", "content.action.edit", "content.code.copy", "versioning", "icons"], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../assets/javascripts/bundle.92b07e13.min.js"></script>
      
        <script src="https://fonts.googleapis.com/css2?family=Fira+Code&amp;family=Inter:wght@400;600&amp;display=swap"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js?config=TeX-MML-AM_CHTML"></script>
      
    
  </body>
</html>