
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A Nextflow-based DNA Methylation Data Analysis Pipeline for Twist NGS data">
      
      
        <meta name="author" content="Jyotirmoy Das">
      
      
      
        <link rel="prev" href="..">
      
      
        <link rel="next" href="../usage/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.18">
    
    
      
        <title>Description - TwistMethylFlow</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.7e37652d.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Inter:300,300i,400,400i,700,700i%7CFira+Code:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Inter";--md-code-font:"Fira Code"}</style>
      
    
    
      <link rel="stylesheet" href="../styles/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="blue-grey" data-md-color-accent="blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#feature-details" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="TwistMethylFlow" class="md-header__button md-logo" aria-label="TwistMethylFlow" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            TwistMethylFlow
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Description
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/JD2112/TwistMethylFlow" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    
  
  
    <li class="md-tabs__item md-tabs__item--active">
      <a href="./" class="md-tabs__link">
        
  
  
    
  
  Description

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../usage/" class="md-tabs__link">
        
  
  
    
  
  Usage

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../credits/" class="md-tabs__link">
        
  
  
    
  
  Credits

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../references/" class="md-tabs__link">
        
  
  
    
  
  References

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="TwistMethylFlow" class="md-nav__button md-logo" aria-label="TwistMethylFlow" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    TwistMethylFlow
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/JD2112/TwistMethylFlow" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Description
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Description
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#feature-details" class="md-nav__link">
    <span class="md-ellipsis">
      Feature Details
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#read-processing" class="md-nav__link">
    <span class="md-ellipsis">
      Read processing
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Read processing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fastqc" class="md-nav__link">
    <span class="md-ellipsis">
      FastQC
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-galore" class="md-nav__link">
    <span class="md-ellipsis">
      Trim Galore
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bismark-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Bismark Analysis
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Bismark Analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#reference-genome-preparation" class="md-nav__link">
    <span class="md-ellipsis">
      Reference Genome Preparation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bismark-alignment" class="md-nav__link">
    <span class="md-ellipsis">
      Bismark Alignment
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bismark-deduplication" class="md-nav__link">
    <span class="md-ellipsis">
      Bismark Deduplication
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bismark-methylation-extractor" class="md-nav__link">
    <span class="md-ellipsis">
      Bismark Methylation Extractor
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bismark-report" class="md-nav__link">
    <span class="md-ellipsis">
      Bismark Report
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#alignment-quality-mapping" class="md-nav__link">
    <span class="md-ellipsis">
      Alignment Quality Mapping
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#qc-reporting" class="md-nav__link">
    <span class="md-ellipsis">
      QC Reporting
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#differential-methylation-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Differential Methylation Analysis
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Differential Methylation Analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#edger" class="md-nav__link">
    <span class="md-ellipsis">
      EdgeR
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#methylkit" class="md-nav__link">
    <span class="md-ellipsis">
      MethylKit
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#post-processing" class="md-nav__link">
    <span class="md-ellipsis">
      Post-processing
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gene-ontology-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Gene Ontology Analysis
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../usage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usage
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../credits/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Credits
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../references/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    References
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<p>TwistMethylFlow integrates various tools and custom scripts to provide a comprehensive analysis workflow for Twist NGS Methylation data.</p>
<table>
<thead>
<tr>
<th>Step</th>
<th>Tool/Software</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>Generate Reference Genome</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/bismark/">Bismark</a></td>
<td>Generate reference genome index files for bismark.</td>
</tr>
<tr>
<td>Raw Data QC</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></td>
<td>Performs quality control checks on raw sequence data.</td>
</tr>
<tr>
<td>Adapter Trimming</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">Trim Galore</a></td>
<td>Trims adapters and low-quality bases from the reads.</td>
</tr>
<tr>
<td>Align Reads</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/bismark/">Bismark (bowtie2)</a></td>
<td>Aligns bisulfite-treated reads to a reference genome.</td>
</tr>
<tr>
<td>Deduplicate removal</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/bismark/">Bismark</a></td>
<td>Removes PCR duplicates from the aligned reads.</td>
</tr>
<tr>
<td>Sort and Indexing</td>
<td><a href="http://www.htslib.org/">Samtools</a></td>
<td>Sorts the aligned and deduplicated BAM files.</td>
</tr>
<tr>
<td>Extract Methylation Calls</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/bismark/">Bismark</a></td>
<td>Extracts methylation calls from the aligned reads.</td>
</tr>
<tr>
<td>Summary Report</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/bismark/">Bismark</a></td>
<td>Generates a summary report of the Bismark alignment and <br>methylation extraction.</td>
</tr>
<tr>
<td>Alignment QC</td>
<td><a href="http://qualimap.conesalab.org/">Qualimap</a></td>
<td>Generates quality control metrics for the aligned reads.</td>
</tr>
<tr>
<td>QC Reporting</td>
<td><a href="https://multiqc.info/">MultiQC</a></td>
<td>Aggregates quality control reports from various steps <br>into a single report.</td>
</tr>
<tr>
<td>Differential Methylation</td>
<td><a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">EdgeR</a>, <br><a href="https://bioconductor.org/packages/release/bioc/html/methylKit.html">MethylKit</a></td>
<td>Performs differential methylation analysis using EdgeR <br>and MethylKit package.</td>
</tr>
<tr>
<td>Post Processing</td>
<td><a href="https://ggplot2.tidyverse.org/">ggplot2</a></td>
<td>Generates summary statistics and visualizations of the <br>differential methylation results.</td>
</tr>
<tr>
<td>GO Analysis</td>
<td><a href="http://geneontology.org/">Gene Ontology</a></td>
<td>Generates GOChord diagram from gene ontology analysis results.</td>
</tr>
</tbody>
</table>
<h2 id="feature-details">Feature Details<a class="headerlink" href="#feature-details" title="Permanent link">&para;</a></h2>
<ol>
<li><strong>READ_PROCESSING:</strong> Checks the quality of raw sequencing data and trims low-quality bases and adapters to improve downstream analysis.</li>
<li><strong>BISMARK_ANALYSIS:</strong> Aligns bisulfite-converted reads to a reference genome, identifies and removes PCR duplicates, sorts and indexes the aligned reads, performs quality control on alignments, and extracts methylation information from the aligned reads.</li>
<li><strong>QC_REPORTING:</strong> Compiles quality control metrics from various steps into a comprehensive report for easy interpretation.</li>
<li><strong>DIFFERENTIAL_METHYLATION:</strong></li>
<li><em>EDGER_ANALYSIS:</em><ul>
<li>Takes coverage files, a design file, and comparison information as input.</li>
<li>Performs differential methylation analysis using the EdgeR Bioconductor package.</li>
<li>Outputs CSV files with differential methylation results for each group comparison. <em>METHYLKIT_ANALYSIS:</em></li>
<li>Takes coverage files, a design file, and comparison information as input.</li>
<li>Performs differential methylation analysis using the EdgeR Bioconductor package.</li>
<li>Outputs CSV files with differential methylation results for each group comparison.</li>
</ul>
</li>
<li><strong>POST_PROCESSING:</strong></li>
<li>Reads the EdgeR results and generates:<ul>
<li>Summary statistics (total DMRs, hyper/hypomethylated regions, significant DMRs)</li>
<li>Volcano plot (visualizing fold change vs. significance)</li>
<li>MA plot (visualizing intensity vs. fold change)</li>
</ul>
</li>
<li>
<p><strong>Functional Analsysis</strong></p>
</li>
<li>
<p>Reads the EdgeR/MethylKit results and generates -&#x20;</p>
</li>
<li>Top <code>n</code> corresponding genes from the EdgeR/MethylKit results are picked up to generate the gene ontology results.</li>
<li>generates a CSV file with the GO classification results (only <strong>Biological Processes).</strong></li>
<li>generates a <strong>Chord diagram</strong> for top 10 results from the GO analysis.</li>
</ol>
<h1 id="read-processing">Read processing<a class="headerlink" href="#read-processing" title="Permanent link">&para;</a></h1>
<p>Read processing subworkflow includes -&#x20;</p>
<ul>
<li>FASTQC - for Quality check of samples</li>
<li>TRIM Galore - adapter trimming</li>
</ul>
<h2 id="fastqc">FastQC<a class="headerlink" href="#fastqc" title="Permanent link">&para;</a></h2>
<p><strong>FASTQC</strong> is a widely used tool for assessing the quality of raw and processed sequencing data. It provides a comprehensive quality check, including metrics like per-base quality scores, GC content, and adapter contamination.</p>
<div class="highlight"><pre><span></span><code>fastqc $args --threads $task.cpus $reads
</code></pre></div>
<p><strong>General Options</strong></p>
<ul>
<li><code>&lt;file{R1,R2}.fastq&gt;</code>: Input FASTQ files (gzip-compressed files, e.g., <code>file1.fastq.gz</code>, are also supported).</li>
<li><code>-o &lt;output_directory&gt;</code>: Specify the directory where reports will be saved. Defaults to the current directory if omitted.</li>
<li><code>-t &lt;number_of_threads&gt;</code>: Specify the number of threads for parallel processing.</li>
</ul>
<details class="note">
<summary>FASTQC Results</summary>
<ol>
<li><strong>HTML Report</strong>: Visual summary of the quality metrics.</li>
<li><strong>ZIP File</strong>: Contains the raw data used to generate the report.</li>
</ol>
<p>Key metrics in the HTML report:</p>
<ul>
<li><strong>Per Base Sequence Quality</strong>:</li>
<li>Boxplots showing quality scores across all positions in reads.</li>
<li>Green indicates high-quality bases (&gt;Q30).</li>
<li><strong>Per Sequence Quality Scores</strong>:</li>
<li>Overall quality of reads in the file.</li>
<li><strong>Per Base GC Content</strong>:</li>
<li>GC content distribution across the length of reads.</li>
<li><strong>Adapter Content</strong>:</li>
<li>Detects overrepresented adapter sequences.</li>
<li><strong>Overrepresented Sequences</strong>:</li>
<li>Identifies frequently occurring sequences (e.g., adapters or contaminants).</li>
</ul>
</details>
<h2 id="trim-galore">Trim Galore<a class="headerlink" href="#trim-galore" title="Permanent link">&para;</a></h2>
<p><strong>Trim Galore</strong> is a versatile tool for trimming sequencing reads and removing adapter sequences. It’s particularly useful for preparing raw sequencing data for downstream applications like alignment or differential expression/methylation analysis. Trim Galore combines the functionalities of <strong>Cutadapt</strong> and <strong>FastQC</strong> for quality control and trimming.</p>
<div class="highlight"><pre><span></span><code>trim_galore --paired --cores $task.cpus $args $reads
</code></pre></div>
<p><strong>General Options</strong></p>
<ul>
<li><code>-q &lt;quality&gt;</code>: Trim low-quality bases from the ends of reads. Default is <code>20</code>.</li>
<li><code>--length &lt;min_length&gt;</code>: Discard reads shorter than the specified length after trimming.</li>
<li><code>--adapter &lt;sequence&gt;</code>: Specify a custom adapter sequence. By default, Trim Galore auto-detects adapters.</li>
<li><code>--gzip</code>: Compress the output files into <code>.gz</code> format.</li>
<li><code>--fastqc</code>: Run <strong>FastQC</strong> before and after trimming.</li>
<li><code>--cores &lt;number&gt;</code>: Use multiple cores for faster processing.</li>
</ul>
<h1 id="bismark-analysis">Bismark Analysis<a class="headerlink" href="#bismark-analysis" title="Permanent link">&para;</a></h1>
<h2 id="reference-genome-preparation">Reference Genome Preparation<a class="headerlink" href="#reference-genome-preparation" title="Permanent link">&para;</a></h2>
<p>Bismark needs to prepare the bisulfite index for the genome.</p>
<ul>
<li>In the current pipeline, user can provide the <code>genome.fasta</code> and the pipeline can index it.</li>
<li>Optinally, user can provide the index files directly, and the pipeline will use it without indexing the genome again.</li>
</ul>
<p><strong>General Options</strong></p>
<ul>
<li><code>--verbose</code>: Prints detailed output during the indexing process.</li>
<li><code>--parallel &lt;threads&gt;</code>: Uses multiple threads to speed up genome preparation.</li>
<li><code>--bowtie2</code>: Specifies that <strong>Bowtie2</strong> will be used for alignment (default option in most versions).</li>
<li><code>--path_to_bowtie &lt;path&gt;</code>: Specifies the path to the Bowtie installation if not in your <code>PATH</code>.</li>
</ul>
<p><strong>Example with Options:</strong></p>
<div class="highlight"><pre><span></span><code>bismark_genome_preparation<span class="w"> </span>--bowtie2<span class="w"> </span>--parallel<span class="w"> </span><span class="m">4</span><span class="w"> </span>&lt;genome.fasta&gt;
</code></pre></div>
<details class="note">
<summary>Reference Genome Preparation</summary>
<p>After successful completion, Bismark generates a bisulfite-converted genome in two orientations (C-&gt;T and G-&gt;A) along with the Bowtie/Bowtie2 indices.</p>
<p>Output directory structure:</p>
<div class="highlight"><pre><span></span><code>your_path_to_reference/
    Bisulfite_Genome/
        CT_conversion/
        GA_conversion/
        genome.1.bt2
        genome.2.bt2
        ...
</code></pre></div>
</details>
<h2 id="bismark-alignment">Bismark Alignment<a class="headerlink" href="#bismark-alignment" title="Permanent link">&para;</a></h2>
<p>This step aligns bisulfite-treated sequencing reads to a reference genome.</p>
<div class="highlight"><pre><span></span><code>bismark --genome &lt;path_to_reference_genome&gt; -1 &lt;reads_R1.fq&gt; -2 &lt;reads_R2.fq&gt; -o &lt;output_directory&gt;
</code></pre></div>
<p><strong>General Options</strong>
  * <code>--genome</code>: Path to the reference genome directory preprocessed with <code>bismark_genome_preparation</code>.
  * <code>-1</code> and <code>-2</code>: Specify paired-end reads. Use <code>-U</code> for single-end reads.
  * <code>-o</code>: Output directory for alignment files.</p>
<details class="note">
<summary>Bismark Alignment</summary>
</details>
<ul>
<li>Produces <code>.bam</code>  alignment files.</li>
<li>Produces <code>.report.txt</code>  and</li>
<li><code>unmapped_reads.fq.gz</code> file.</li>
</ul>
<h2 id="bismark-deduplication">Bismark Deduplication<a class="headerlink" href="#bismark-deduplication" title="Permanent link">&para;</a></h2>
<p>This step removes duplicate reads to avoid overestimating methylation levels.</p>
<div class="highlight"><pre><span></span><code>deduplicate_bismark ${paired_end} $args --bam $bam
</code></pre></div>
<p><strong>General options:</strong>
  * <code>args</code> options use all arguments from bismark deduplicate command.
  * <code>--paired</code>: Use this for paired-end data. Remove for single-end reads.
  * <code>--bam</code>: Specifies the input BAM file, generated from <code>bismark alignment</code></p>
<details class="note">
<summary>Bismark Deduplicate Removal</summary>
</details>
<ul>
<li>Generates a deduplicated <code>.bam</code> file.</li>
<li>Produces <code>deduplicated_report.txt</code> file.</li>
</ul>
<h2 id="bismark-methylation-extractor">Bismark Methylation Extractor<a class="headerlink" href="#bismark-methylation-extractor" title="Permanent link">&para;</a></h2>
<p>Extract methylation data from deduplicated BAM files.</p>
<div class="highlight"><pre><span></span><code>bismark_methylation_extractor<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--bedGraph<span class="w"> </span>--gzip<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-o<span class="w"> </span>&lt;output_directory&gt;<span class="w"> </span>&lt;deduplicated.bam&gt;
</code></pre></div>
<p><strong>General options:</strong>
  * <code>--bedGraph</code>: Generates bedGraph file
  * <code>--gzip</code>: Compresses the output files.</p>
<details class="note">
<summary>Bismark Methylation Extractor</summary>
</details>
<ul>
<li>Generates <code>.bismark.cov.gz</code> files and methylation call data in CpG, CHG, and CHH contexts.</li>
<li>Generates <code>bedGraph.gz</code> file.</li>
<li>Also generates <code>splitting_report.txt</code> file.</li>
</ul>
<h2 id="bismark-report">Bismark Report<a class="headerlink" href="#bismark-report" title="Permanent link">&para;</a></h2>
<p>Generate a summary report of alignment and methylation statistics.</p>
<p><strong>Command:</strong></p>
<div class="highlight"><pre><span></span><code>bismark2report
</code></pre></div>
<details class="note">
<summary>Bismark Report</summary>
<p>Produces an HTML file summarizing:
    * Alignment efficiency.
    * Duplicate rates.
    * Methylation levels (CpG, CHG, CHH contexts).</p>
</details>
<h1 id="alignment-quality-mapping">Alignment Quality Mapping<a class="headerlink" href="#alignment-quality-mapping" title="Permanent link">&para;</a></h1>
<p>The main module for assessing alignment quality is <code>qualimap bamqc</code>.</p>
<div class="highlight"><pre><span></span><code>qualimap<span class="w"> </span>bamqc<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-bam<span class="w"> </span>&lt;input.bam&gt;<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-outdir<span class="w"> </span>&lt;output_directory&gt;<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-outformat<span class="w"> </span>&lt;html&gt;<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--java-mem-size
</code></pre></div>
<p><strong>General Options:</strong>
  * <code>-bam &lt;input.bam&gt;</code>: Path to the aligned BAM file (e.g., deduplicated BAM file).
  * <code>-outdir &lt;output_directory&gt;</code>: Directory for output reports.
  * <code>-outformat &lt;pdf|html&gt;</code>: Choose the output format for the report.</p>
<details class="note">
<summary>Alignment Quality Check</summary>
<p>The output includes:</p>
<ol>
<li><strong>General Alignment Statistics</strong>:</li>
<li>Total number of reads.</li>
<li>Percentage of mapped reads.</li>
<li>Percentage of properly paired reads (for paired-end data).</li>
<li><strong>Coverage Statistics</strong>:</li>
<li>Mean coverage depth.</li>
<li>Percentage of the genome covered at varying depths (e.g., 1x, 5x, 10x).</li>
<li><strong>Insert Size Distribution</strong> (for paired-end reads):</li>
<li>Provides insights into library preparation and sequencing quality.</li>
<li><strong>GC Content Distribution</strong>:</li>
<li>Checks for bias in GC content distribution compared to expected values.</li>
<li><strong>Read Quality Metrics</strong>:</li>
<li>Distribution of mapping quality scores.</li>
</ol>
</details>
<h1 id="qc-reporting">QC Reporting<a class="headerlink" href="#qc-reporting" title="Permanent link">&para;</a></h1>
<p><em>MultiQC</em> is used for the QC reporting combining all results from the <em>FastQC, Trim galore, Bismark Alignment, Bismark Deduplication, Bismark summary report,</em> and <em>Qualimap results.</em></p>
<p><strong>Output</strong>:</p>
<ul>
<li>Generates an interactive HTML report (<code>multiqc_report.html</code>) and a data file (<code>multiqc_data.json</code>).</li>
<li>Output includes summary statistics, plots, and tool-specific metrics.</li>
</ul>
<h1 id="differential-methylation-analysis">Differential Methylation Analysis<a class="headerlink" href="#differential-methylation-analysis" title="Permanent link">&para;</a></h1>
<p>To calculate the differential methylation from the input samples, two different methods can be used -&#x20;</p>
<ul>
<li><a href="#edger">EdgeR</a> (Default) or</li>
<li><a href="#methylkit">MethylKit</a></li>
</ul>
<h2 id="edger">EdgeR<a class="headerlink" href="#edger" title="Permanent link">&para;</a></h2>
<p><strong>edgeR</strong> is a Bioconductor package primarily used for RNA-seq differential expression analysis but can also handle differential methylation analysis when paired with bisulfite sequencing data. This requires pre-processed methylation data, such as counts of methylated (<code>M</code>) and unmethylated (<code>U</code>) reads at each cytosine position or region of interest.</p>
<div class="highlight"><pre><span></span><code>Rscript $baseDir/bin/edgeR_analysis.R \
        --coverage_files &#39;${coverage_files}&#39; \
        --design &quot;${design_file}&quot; \
        --compare &quot;${compare_str}&quot; \
        --output . \
        --threshold ${coverage_threshold}
</code></pre></div>
<p><strong>General options</strong></p>
<ul>
<li><code>--coverage_files:</code> selected from the <code>bismark_methylation_extractor</code> files.</li>
<li><code>--design</code>: selected from the <code>Sample_sheet.csv</code>&#x20;</li>
<li><code>--compare</code>: selected from the <code>Sample_sheet.csv</code> .</li>
</ul>
<details class="note">
<summary>EdgeR Results</summary>
<ul>
<li>Generates <code>EdgeR_group_&lt;compare_str&gt;.csv</code>.</li>
</ul>
</details>
<h2 id="methylkit">MethylKit<a class="headerlink" href="#methylkit" title="Permanent link">&para;</a></h2>
<p><strong>MethylKit</strong> is an R package designed for analyzing bisulfite sequencing data, particularly for differential methylation analysis. It supports genome-wide methylation data and is ideal for CpG, CHH, and CHG methylation studies.</p>
<div class="highlight"><pre><span></span><code>Rscript $baseDir/bin/run_methylkit.R \
        --coverage_files &#39;${coverage_files}&#39; \
        --design &quot;${design_file}&quot; \
        --compare &quot;${compare_str}&quot; \
        --output . \
        --threshold ${coverage_threshold}
</code></pre></div>
<p><strong>General options</strong></p>
<ul>
<li><code>--coverage_files:</code> selected from the <code>bismark_methylation_extractor</code> files.</li>
<li><code>--design</code>: selected from the <code>Sample_sheet.csv</code>&#x20;</li>
<li><code>--compare</code>: selected from the <code>Sample_sheet.csv</code> .</li>
</ul>
<details class="note">
<summary>MethylKit Results</summary>
<ul>
<li>Generates <code>Methylkit_group_&lt;compare_str&gt;.csv</code> </li>
</ul>
</details>
<h1 id="post-processing">Post-processing<a class="headerlink" href="#post-processing" title="Permanent link">&para;</a></h1>
<p>Generates A) <strong>Volcano Plot,</strong> B) <strong>MA Plot</strong> and C) <strong>Summary Statistics</strong> from the Diffrential Methylation results.</p>
<details class="note">
<summary>Figures</summary>
<p><strong>Volcano Plot</strong>
<img alt="Volcano Plot" src="../images/volcano_plot.png" /></p>
<p><strong>MA plot</strong>
<img alt="MA plot" src="../images/ma_plot.png" /></p>
</details>
<h1 id="gene-ontology-analysis">Gene Ontology Analysis<a class="headerlink" href="#gene-ontology-analysis" title="Permanent link">&para;</a></h1>
<p>The pipeline has also a module to perform the Gene Ontology analysis from the top <code>n</code> corresponding genes from the differential methylation results (EdgeR/MethylKit) using the <em>clusterProfiler</em> package.</p>
<p>The results generates a full table with all <em>Biological Processes</em> and a <em>Chord diagram</em> with top 10 functions identified in the analysis.</p>
<details class="note">
<summary>GOchord Diagram</summary>
<p><img alt="GOchord Diagram" src="../images/goplot.png" /></p>
</details>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
  <div style="
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
    font-size: 1.5em;
    padding: 1em 2em;
    border-top: 1px solid #000000;
    color: #ffffff;
    text-align: center;
    background-color: #000000;
  ">
    <div style="flex: 1; text-align: left;">
        © Jyotirmoy Das 2025 
    </div>
    <div style="flex: 1; text-align: center;">
      <code style="font-size: 0.8em;">
        3D1F C87F 8D52 BFD4 5A1D <br> 3C86 63F2 2E14 6A0A 6B98
      </code>
    </div>
    <div style="flex: 1; text-align: right;">
      Made with <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" style="color: #ffffff;">MkDocs</a>
    </div>
  </div>

    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["navigation.instant", "navigation.tracking", "navigation.expand", "navigation.sections", "navigation.top", "content.code.annotate", "content.tabs.link", "navigation.tabs", "search.suggest", "search.highlight", "toc.integrate", "content.action.edit", "content.code.copy", "versioning"], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../assets/javascripts/bundle.92b07e13.min.js"></script>
      
        <script src="https://fonts.googleapis.com/css2?family=Fira+Code&amp;family=Inter:wght@400;600&amp;display=swap"></script>
      
    
  </body>
</html>