# Citations

### TwistMethylFlow
>**Das, J.** (2024). *TwistMethylFlow (v1.0.0)*. Zenodo.  [https://doi.org/10.5281/zenodo.14204261](https://doi.org/10.5281/zenodo.14204261)

## Pipeline tools

### FastQC
>Andrews, S. (2010). FastQC: A Quality Control Tool for High Throughput Sequence Data [Online].

### Trim Galore
>Felix Krueger, Frankie James, Phil Ewels, Ebrahim Afyounian, Michael Weinstein, Benjamin Schuster-Boeckler, Gert Hulselmans, & sclamons. (2023). FelixKrueger/TrimGalore: v0.6.10 - add default decompression path (0.6.10). Zenodo. https://doi.org/10.5281/zenodo.7598955

### Bismark
>**Krueger, F.**, & **Davies, R.** (2019). *Bismark: a versatile tool for aligning sequencing data to bisulfite-converted genomes*. Bioinformatics, 35(4), 623-625. [https://doi.org/10.1093/bioinformatics/bty724](https://doi.org/10.1093/bioinformatics/bty724)

### SAMtools
>Danecek P, Bonfield JK, Liddle J, et al. Twelve years of SAMtools and BCFtools. GigaScience. 2021;10(2):giab008. doi:10.1093/gigascience/giab008

### QualiMap
>Konstantin Okonechnikov, Ana Conesa, Fernando García-Alcalde, Qualimap 2: advanced multi-sample quality control for high-throughput sequencing data, Bioinformatics, Volume 32, Issue 2, January 2016, Pages 292–294, [https://doi.org/10.1093/bioinformatics/btv566](https://doi.org/10.1093/bioinformatics/btv566)

### MultiQC
>Ewels P, Magnusson M, Lundin S, Käller M. MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics. 2016 Oct 1;32(19):3047-8. doi: 10.1093/bioinformatics/btw354. Epub 2016 Jun 16. PubMed PMID: 27312411; PubMed Central PMCID: PMC5039924.

### EdgeR
>Chen Y, Chen L, Lun ATL, Baldoni P, Smyth GK (2025). “edgeR v4: powerful differential analysis of sequencing data with expanded functionality and improved support for small counts and larger datasets.” Nucleic Acids Research, 53(2), gkaf018. [doi:10.1093/nar/gkaf018](doi:10.1093/nar/gkaf018).

### MethylKit
>**Morgan, M., & Huber, W.** (2020). *methylKit: a package for the analysis of DNA methylation data in R*. Bioinformatics, 36(1), 214-216.  [https://doi.org/10.1093/bioinformatics/btz262](https://doi.org/10.1093/bioinformatics/btz262)

### ggplot2
>Wickham H (2016). ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York. ISBN 978-3-319-24277-4, [https://ggplot2.tidyverse.org](https://ggplot2.tidyverse.org).

### Gene Ontology
>Xu S, Hu E, Cai Y, Xie Z, Luo X, Zhan L, Tang W, Wang Q, Liu B, Wang R, Xie W, Wu T, Xie L, Yu G (2024). “Using clusterProfiler to characterize multiomics data.” Nature Protocols, 19(11), 3292-3320. doi:10.1038/s41596-024-01020-z, [https://www.nature.com/articles/s41596-024-01020-z](https://www.nature.com/articles/s41596-024-01020-z).

>Walter, Wencke, Fátima Sánchez-Cabo, and Mercedes Ricote. "GOplot: an R package for visually combining expression data with functional analysis." Bioinformatics (2015): btv300.

## Software packaging/containerisation tools

### Anaconda
>Anaconda Software Distribution. Computer software. Vers. 2-2.4.0. Anaconda, Nov. 2016. Web.

### Bioconda
>Grüning B, Dale R, Sjödin A, Chapman BA, Rowe J, Tomkins-Tinch CH, Valieris R, Köster J; Bioconda Team. Bioconda: sustainable and comprehensive software distribution for the life sciences. Nat Methods. 2018 Jul;15(7):475-476. doi: 10.1038/s41592-018-0046-7. PubMed PMID: 29967506.

### BioContainers
>da Veiga Leprevost F, Grüning B, Aflitos SA, Röst HL, Uszkoreit J, Barsnes H, Vaudel M, Moreno P, Gatto L, Weber J, Bai M, Jimenez RC, Sachsenberg T, Pfeuffer J, Alvarez RV, Griss J, Nesvizhskii AI, Perez-Riverol Y. BioContainers: an open-source and community-driven framework for software standardization. Bioinformatics. 2017 Aug 15;33(16):2580-2582. doi: 10.1093/bioinformatics/btx192. PubMed PMID: 28379341; PubMed Central PMCID: PMC5870671.

### Docker
>Merkel, D. (2014). Docker: lightweight linux containers for consistent development and deployment. Linux Journal, 2014(239), 2. doi: 10.5555/2600239.2600241.

### Nextflow

>Di Tommaso P, Chatzou M, Floden EW, Barja PP, Palumbo E, Notredame C. Nextflow enables reproducible computational workflows. Nat Biotechnol. 2017 Apr 11;35(4):316-319. doi: 10.1038/nbt.3820. PubMed PMID: 28398311.

### R and Bioconductor
>R. Ihaka and R. Gentleman (1996), R: A Language for Data Analysis and Graphics, Journal of Computational and Graphical Statistics, Vol 5, No 3, 299-314.

>Huber, W., Carey, V., Gentleman, R. et al. Orchestrating high-throughput genomic analysis with Bioconductor. Nat Methods 12, 115–121 (2015). [https://doi.org/10.1038/nmeth.3252](https://doi.org/10.1038/nmeth.3252).

### Singularity
>Kurtzer GM, Sochat V, Bauer MW. Singularity: Scientific containers for mobility of compute. PLoS One. 2017 May 11;12(5):e0177459. doi: 10.1371/journal.pone.0177459. eCollection 2017. PubMed PMID: 28494014; PubMed Central PMCID: PMC5426675.

---

## 👤 Authors and Maintainers

| Role         | Name & Link                                                                 | Description                                                                                      |
|--------------|------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------|
| **Main Author** | [Jyotirmoy Das](https://github.com/JD2112)                                  | Lead designer, developer, and overall coordinator for the **TwistMethylFlow** pipeline.           |
| **Maintainer**  | [Jyotirmoy Das](https://github.com/JD2112)                                  | Actively maintaining the pipeline, fixing bugs, and implementing new features.                  |
| **Contributors**| [All Contributors](https://github.com/JD2112/TwistMethylFlow/pulls)           | Thanks to everyone who helped in development, testing, and enhancement. Contributions welcome!  |

---

## 📄 Citation

If you use **TwistMethylFlow** in your work, please cite:

> Das, J. (2024). *TwistMethylFlow (v1.0.0)*. Zenodo. [https://doi.org/10.5281/zenodo.14204261](https://doi.org/10.5281/zenodo.14204261)

---

## ❓ Help, FAQ, Troubleshooting

- Refer to the [User Manual](index.md) for detailed usage instructions and guidance.
- For bug reports, questions, or feature requests, please open an [issue on GitHub](https://github.com/JD2112/TwistMethylFlow/issues).

---

## 📜 License

This project is licensed under the terms of the **GNU General Public License v3.0**.  

---

## 🙏 Acknowledgements

### Research Institutions

- **Core Facility, Faculty of Medicine and Health Sciences, Linköping University**  
  [Linköping University Core Facility](https://liu.se/en/organisation/liu/medfak/coref)

- **Clinical Genomics Linköping, Science for Life Laboratory, Sweden**  
  [Clinical Genomics Linköping](https://www.scilifelab.se/units/clinical-genomics-linkoping/)

- **National Academic Infrastructure for Supercomputing in Sweden (NAISS)**
  > The computations/data storage were enabled by resources provided by the National Academic Infrastructure for Supercomputing in Sweden (NAISS), partially funded by the Swedish Research Council through grant agreement no. 2022-06725.

These institutions provided invaluable resources and support for the development and testing of the TwistMethylFlow pipeline.

### Additional Thanks

- **Open Source Community**: Special thanks to the developers and contributors of [Nextflow](https://www.nextflow.io/), [Bismark](https://www.bioinformatics.babraham.ac.uk/projects/bismark/), [methylKit](https://github.com/observer101/methylKit), [EdgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html) and other bioinformatics tools integrated into this pipeline.
- **Users and Testers**: A heartfelt thank you to all the early adopters, testers, and collaborators who helped refine and improve the tool through their feedback and bug reports.

---
